website_attribute_set/controllers/main.py

```diff
@@ -0,0 +1,421 @@
+# Copyright 2011 Kencove (http://www.kencove.com).
+# @author Mohamed Alkobrosli <malkobrosly@kencove.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+
+from werkzeug.exceptions import NotFound
+
+from odoo import fields
+from odoo.http import request, route
+from odoo.models import BaseModel
+from odoo.osv import expression
+from odoo.tools import SQL, float_round, groupby, lazy
+
+from odoo.addons.website.controllers.main import QueryURL
+from odoo.addons.website_sale.controllers import main
+
+
+class WebsiteSale(main.WebsiteSale):
+    @route(
+        [
+            "/shop",
+            "/shop/page/<int:page>",
+            '/shop/category/<model("product.public.category"):category>',
+            '/shop/category/<model("product.public.category"):category>/page/<int:page>',
+        ],
+        type="http",
+        auth="public",
+        website=True,
+        sitemap=main.WebsiteSale.sitemap_shop,
+    )
```

Comment:
```suggestion
    @route()
```

> the decorators of all methods are combined, if the overriding method's decorator has no argument all previous ones will be kept, any provided argument will override previously defined ones e.g.:

see in: https://www.odoo.com/documentation/18.0/vi/developer/reference/backend/http.html#controllers

Reply:
thanks, I didn't know that I can add route decorator empty there.